### ISSUE TITLE
Ruby rejects `tap{|;_1|}` but YARP accepts it

### ISSUE THREAD

**Issue by tompng** [ruby/prism#1579]
Similar to #1546, same for `->(;_1){}`
```ruby
irb(main):001> eval 'tap{|;_1|}'
(irb):1:in `eval': (eval):1: _1 is reserved for numbered parameter (SyntaxError)
	from (irb):1:in `<main>'
	from bin/console:8:in `<main>'
irb(main):002> YARP.parse 'tap{|;_1|}'
=> 
#<YARP::ParseResult:0x00000001047fa0a8
 @comments=[],
 @errors=[],
 @source=#<YARP::Source:0x0000000104678838 @offsets=[0], @source="tap{|;_1|}">,
 @value=
  @ ProgramNode (location: (1,0)-(1,10))
  ├── locals: []
  └── statements:
      @ StatementsNode (location: (1,0)-(1,10))
      └── body: (length: 1)
          └── @ CallNode (location: (1,0)-(1,10))
              ├── receiver: ∅
              ├── call_operator_loc: ∅
              ├── message_loc: (1,0)-(1,3) = "tap"
              ├── opening_loc: ∅
              ├── arguments: ∅
              ├── closing_loc: ∅
              ├── block:
              │   @ BlockNode (location: (1,3)-(1,10))
              │   ├── locals: [:_1]
              │   ├── parameters:
              │   │   @ BlockParametersNode (location: (1,4)-(1,9))
              │   │   ├── parameters: ∅
              │   │   ├── locals: (length: 1)
              │   │   │   └── @ BlockLocalVariableNode (location: (1,6)-(1,8))
              │   │   │       └── name: :_1
              │   │   ├── opening_loc: (1,4)-(1,5) = "|"
              │   │   └── closing_loc: (1,8)-(1,9) = "|"
              │   ├── body: ∅
              │   ├── opening_loc: (1,3)-(1,4) = "{"
              │   └── closing_loc: (1,9)-(1,10) = "}"
              ├── flags: ∅
              └── name: "tap",
 @warnings=[]>
```

**Comment by tompng** [comment]
YARP also accepts these
```ruby
tap{|a;a,_1,a!,a?|}
->(a;a,_1,a!,a?){}
```

**Comment by tompng** [comment]
Looks like it's fixed in #1588 and #1628
